crates/sui-framework/sources/governance/delegation.move

```diff
@@ -191,7 +191,7 @@ module sui::delegation {
             self.next_reward_unclaimed_epoch <= epoch_to_claim
         } else {
             let ending_epoch = *option::borrow(&self.ending_epoch);
-            ending_epoch > epoch_to_claim
+            ending_epoch > epoch_to_claim && self.next_reward_unclaimed_epoch <= epoch_to_claim
```

Comment:
somehow dedup with line 191?

Reply:
Gave it a try. Hopefully it's cleaner now.